plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -640,6 +645,28 @@ else if (value instanceof Document) {
                 typeSignature = new TypeSignature(StandardTypes.ROW, parameters);
             }
         }
+        else if (value instanceof DBRef) {
+            List<TypeSignatureParameter> parameters = new ArrayList<>();
+
+            Optional<TypeSignature> idFieldType = guessFieldType(((DBRef) value).getId());
+            if (idFieldType.isPresent()) {
+                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(ID)), idFieldType.get())));
+            }
+
+            Optional<TypeSignature> dbNameFieldType = guessFieldType( ((DBRef) value).getDatabaseName());
```

Comment:
I suppose we don't need to guess field types for database name and collection name of `DBRef` since they're always string. Do you have any reason to use `guessFieldType` method?

Reply:
I agree. Kind of thought to do it for consistency. Have made the changes.